media-source-respec.html

```diff
@@ -757,38 +766,70 @@ <h4><dfn>Attaching to a media element</dfn></h4>
+          and current media resource in the local mode logic in the [=resource fetch algorithm=]. This also
+          means that the remote mode logic that includes observance of any preload attribute is skipped when a
+          MediaSource object is attached. Even with that eventual change to [[HTML]], the execution of the following
+          steps at the beginning of the local mode logic is still required when the current media resource is a
+          MediaSource object.</p>
+          <p class="note">Relative to the action which triggered the media element's resource selection algorithm, these
+          steps are asynchronous. The [=resource fetch algorithm=] is run after the task that invoked the resource
+          selection algorithm is allowed to continue and a stable state is reached. Implementations may delay the steps
+          in the "<i>Otherwise</i>" clause, below, until the MediaSource object is ready for use.</p>
           <dl class="switch">
-            <dt>If the media provider object is
-              a URL record whose object is a {{MediaSource}} that was constructed in a {{DedicatedWorkerGlobalScope}},
-              such as would occur if attempting to use  a [=MediaSource object URL=] from a
-              {{DedicatedWorkerGlobalScope}} {{MediaSource}}
+            <dt>If the media provider object is a URL record whose object is a {{MediaSource}} that was constructed in a
+              {{DedicatedWorkerGlobalScope}}, such as would occur if attempting to use a [=MediaSource object URL=]
+              from a {{DedicatedWorkerGlobalScope}} {{MediaSource}}
               </dt>
-            <dd>Run the <a def-id="media-data-cannot-be-fetched"></a> steps of the <a def-id="resource-fetch-algorithm"></a>'s <a def-id="media-data-processing-steps-list"></a>.</dd>
-            <dt>If {{MediaSource/readyState}} is NOT set to {{ReadyState/""closed""}} 
-            <dd>Run the <a def-id="media-data-cannot-be-fetched"></a> steps of the <a def-id="resource-fetch-algorithm"></a>'s <a def-id="media-data-processing-steps-list"></a>.</dd>
+            <dd>Run the <a def-id="media-data-cannot-be-fetched"></a> steps of the [=resource fetch algorithm=]'s
+              <a def-id="media-data-processing-steps-list"></a>.
+              <div class="note">This prevents using [=MediaSource object URLs=] for DedicatedWorker MediaSource
+                attachments. Transferring {{MediaSource}}.{{MediaSource/handle}} from the DedicatedWorker to the Window
+                context and assigning it to the media element's {{HTMLMediaElement/srcObject}} attribute is the only way
+                to attach such a MediaSource.
+              </div>
+            </dd>
+            <dt>If the media provider object is a {{MediaSourceHandle}} whose underlying {{MediaSource}} has ever
+              been used beyond this point by in any {{HTMLMediaElement}}'s [=resource fetch algorithm=]</dt>
```

Comment:
"been used beyond this point" is then implicit hidden state of the object. Isn't there some explicit state that could be used here ? Could we have an [[attached]] internal slot ?

Reply:
As on observation, there is a kind of inconsistency in that [[has ever been assigned as srcobject]] is used to block transfer, but that flag is not used to block setting the same handle on another `srcObject`.  I think your design here is reasonable, however, with your approach of erroring (later) during the resource fetch algorithm when the same handle is assigned twice, because that is where other kinds of resources fail to load.

A distinct slot would help clarify the distinction.

Reply:
> "been used beyond this point" is then implicit hidden state of the object. Isn't there some explicit state that could be used here ? Could we have an [[attached]] internal slot ?

We could add something like [[has_ever_been_attached]] to a MediaSource object, and use it to condition this worker/srcObject/Handle-specific attachment logic to prevent attachment via a handle from succeeding if the underlying MediaSource is or has ever been attached, even if that MediaSource is a Window MediaSource that was previously (or is still currently) attached using an object URL.

The intent here is not only to say that a currently-attached underlying MediaSource